pandapower/test/loadflow/test_scenarios.py

```diff
@@ -507,5 +507,33 @@ def test_transformer_with_two_open_switches():
     assert net.res_trafo.vm_lv_pu.at[t] != net.res_bus.vm_pu.at[b2]
 
 
+def test_motor():
+    net = pp.create_empty_network()
+    b1 = pp.create_bus(net, 0.4)
+    b2 = pp.create_bus(net, 0.4)
+    pp.create_line(net, b1, b2, length_km=0.1, std_type="NAYY 4x50 SE")
+    pp.create_ext_grid(net, b1)
+    p_mech = 0.1
+    cos_phi = 0.98
+    efficiency = 95
+    pp.create_motor(net, b2, pn_mech_mw=0.1, cos_phi=cos_phi,
+                    efficiency_percent=efficiency)
+
+    pp.runpp(net)
+    p = net.res_motor.p_mw.iloc[0]
+    q = net.res_motor.q_mvar.iloc[0]
+    s = np.sqrt(p**2+q**2)
+    assert p == p_mech / efficiency * 100
+    assert p/s == cos_phi
+    res_bus_motor = net.res_bus.copy()
+
+    pp.create_load(net, b2, p_mw=net.res_motor.p_mw.values[0],
+                   q_mvar=net.res_motor.q_mvar.values[0])
+    net.motor.in_service = False
+
+    pp.runpp(net)
+    assert net.res_bus.equals(res_bus_motor)
```

Comment:
Isn't this test a bit weak? It is not clear to me how it could ever fail. Can you compare with PowerFactory also here?

Reply:
This just tests that all power values are internally added in the correct way with the correct sign to the power demand vector and results are correctly written in the res_motor dataframe (which is not automatically given).

You are correct that it does not test the motor model itself. I tried to validate against PowerFactory, but they have a much more complex model that works with slip curves, rotor impedances etc., so  was not able to reproduce this more simple model. Do you have an idea for a software to validate against?